tests/test_CaseInsensitve.py

```diff
@@ -93,45 +105,69 @@ def test_equal(self):
         d = CaseInsensitiveDict({"opTIon1": value1})
         self.assertEqual(d, self.d)
 
+    def test_pickle(self):
+        new_dict = pickle.loads(pickle.dumps(self.d))
+        self.assertEqual(self.d, new_dict)
+
+    def test_items(self):
+        res = []
+        for k, v in self.d2.items():
+            res.append((k, v))
+        self.assertEqual(res, [("opTION1", value2), ("optioN2", value2)])
+
+    def test_iter(self):
```

Comment:
I suggest we add a test also using the `keys()` and `values()` methods, since that is somewhat used.

Reply:
Yep agreed, I have added those.